src/main/java/com/example/survey/models/Question.java

```diff
@@ -10,7 +11,9 @@
     protected long id;
     protected String question;
 
-    public abstract String getAnswer(); // setter handled by subclass
+    public abstract Set<String> getAnswers();
```

Comment:
Is this when your suppose to use a generic? Instead of `String` since like you said, classes have different types of answers

Reply:
I did some reading on these 'generics', I thought I would share. As a go developer, I have never heard of this language feature before.

A generic is being used here. Type `Set` takes a generic, that is what `<T>` means in java. The java compiler replaces generics with types at compile time, using a technique that involves [type erasure.](https://javapapers.com/core-java/type-erasure/). You cannot put a wildcard class into Set here to make it have multiple types in it, but you can use polymorphism to achieve something similar.

The type that represents the answer needs to be able to:
 * Be displayed as a String on the frontend for all `Question` subclasses
 * Have a numerical value for the `RangeQuestion` specifically

Pretty straightforward right? I don't think using a generic will make things simpler here.

I think you mean one of:
1. Make the `Question` class have a generic like `Question<T extends Object>` (which like I said, I think is needlessly complicated)
1. Put an interface here like `Printable` or `Comparable`

---
1. This is bad for this solution. A Question should not have its result set type be generic, because for all questions, we know what their result type should be ahead of time. This knowledge is defined with code in each subclass.`TextQuestion` is `String`. `RangeQuestion` is `Integer`, and `OptionQuestion` is `String`. Also, the constraint on the type is that it can be printed, and for the `RangeQuestion`, it needs to be able to be compared with `>=<`. The type constraint if `Question` was a generic class would be `<T extends Object>` for the `toString()` that `Object` provides. We could then use `Integer.parseInt()`. This is actually just a more complicated, and needlessly cluttered way to do polymorphism. If you use generics and the constraint is any `Object`, which every class subclasses, it is just polymorphism with extra steps.

1. Yes, this would be a good place for an interface like `Comparable`. But not in this case. The functions called on the answer type would be `toString()`, and in one specific case, `greaterThan(int i)...`. However, hibernate does not play well with collections of interfaces. It is hard to know their size, and what fields they have. I didn't spend that much time on this solution, as there is a clearly better solution:

1. Making this a `Result` class. `Result` holds both a `String` and an `int`. It has methods for setting its value that will be used by the specific class implementation. This is what I have now, but the code for setting and parsing would be contained in its own class.

Reply:
go developers be like